### PR TITLE
[APB-2231]: Filter Active Relationships that have a set end date

### PR DESCRIPTION
**Pull request by JTY009**

### CODE REVIEW

File: app/uk/gov/hmrc/agentclientrelationships/controllers/RelationshipsController.scala [hunk 1]
```diff
@@ -166,14 +166,16 @@ class RelationshipsController @Inject() (
 
   def getItsaRelationship: Action[AnyContent] = AuthorisedAsItSaClient { implicit request => clientId =>
     service.getItsaRelationshipForClient(clientId).map {
-      case Some(relationship) => Ok(Json.toJson(relationship))
+      case Some(relationship) if relationship.endDate.toString() == "9999-12-31" => Ok(Json.toJson(relationship))
```
Comment (arturopala) [inline app/uk/gov/hmrc/agentclientrelationships/controllers/RelationshipsController.scala]: why not `!relationship.endDate.isAfter(LocalDate.now())`?

File: app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala [hunk 2]
```diff
@@ -43,22 +47,24 @@ object MtdItIdBusinessDetails {
...
 
-  implicit val reads: Reads[ItsaRelationship] =
-    (JsPath \ "agentReferenceNumber").read[Arn].map(arn => ItsaRelationship(arn))
+  implicit val reads: Reads[ItsaRelationship] = (
+    (JsPath \ "agentReferenceNumber").read[Arn] and
+    (JsPath \ "dateTo").read[LocalDate])(ItsaRelationship.apply _)
 }
 
-case class VatRelationship(arn: Arn)
+case class VatRelationship(arn: Arn, endDate: LocalDate)
```
Comment (arturopala) [inline app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala]: should `endDate` be optional?

File: app/uk/gov/hmrc/agentclientrelationships/controllers/RelationshipsController.scala (hunk 1 above)
Comment (JTY009) [reply app/uk/gov/hmrc/agentclientrelationships/controllers/RelationshipsController.scala]: Ah that sounds better

File: app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala (hunk 2 above)
Comment (JTY009) [reply app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala]: could be the same argument for ARN and further down the line converts the NotFoundException to None -->. https://github.com/hmrc/agent-client-relationships/blob/master/app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala#L136

Comment (arkowalski) [reply app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala]: the endDate default value is 9999-12-31 when creating the relationship, and gets updated when deleted. So I gathered from Paul

**Pull request merged**